### PR TITLE
add better null support resolves #4

### LINKED ISSUE #4: piping a list with null values will throw an error
looks like a parameter binding issue.

    Cannot validate argument on parameter 'InputObject'. The argument is null or empty.

example input

    $list = @($obj1, $obj2, $null, $null, $obj3)

### PR DESCRIPTION
**Pull request by KevinMarquette**
removed parameter validation that was looking for null values and added tests

### CODE REVIEW

File: Tests/Resolve-DependencyOrder.Tests.ps1 [hunk 1]
```diff
@@ -73,4 +73,14 @@ Describe "function Resolve-DependencyOrder" -Tag Build {
...
-}
\ No newline at end of file
+
+    It "Handles null items in a list" {
+        $list = @(1, 2, $null, , 3)
+        $results = Resolve-DependencyOrder -InputObject $list -Key {$_} -DependsOn {$null}
+    }
+
+    It "Handles null items in a list over the pipe" {
+        $list = @(1, 2, $null, , 3)
```
Comment (EklipZgit) [inline Tests/Resolve-DependencyOrder.Tests.ps1:83]: need a test for if the list itself is null, or no?

**Review by EklipZgit — APPROVED** [review]
